src/lib/redux-cozy-api.js

```diff
@@ -0,0 +1,484 @@
+      return state
+  }
+}
+
+const hasMore = (state = false, action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      const response = action.response
+      return response.next !== undefined ? response.next : state
+    default:
+      return state
+  }
+}
+
+const count = (state = 0, action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      const response = action.response
+      return response.meta && response.meta.count ? response.meta.count : response.data.length
+    case ADD_REFERENCED_FILES:
+      return state + action.ids.length
+    case REMOVE_REFERENCED_FILES:
+      return state - action.ids.length
+    default:
+      return state
+  }
+}
+
+const ids = (state = [], action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      if (!action.skip) {
+        return action.response.data.map(doc => doc.id)
+      }
+      return [
+        ...state,
+        ...action.response.data.map(doc => doc.id)
+      ]
+    case ADD_REFERENCED_FILES:
+      // TODO: here we don't follow JSONAPI conventions... (we pass an array of ids in the action)
+      return [
+        ...state,
+        ...action.ids
+      ]
+    case REMOVE_REFERENCED_FILES:
+      return state.filter(id => action.ids.indexOf(id) === -1)
+    case RECEIVE_CREATION_CONFIRM:
+      return [
+        ...state,
+        action.response.data[0].id
+      ]
+    case RECEIVE_DELETION_CONFIRM:
+      return state.filter(id => id !== action.response.data[0].id)
+    default:
+      return state
+  }
+}
+
+const endpoint = combineReducers({
+  type,
+  fetchStatus,
+  lastFetch,
+  hasMore,
+  count,
+  ids
+})
+
+const endpoints = (state = {}, action) => {
+  switch (action.type) {
+    case FETCH_DOCUMENTS:
+    case FETCH_REFERENCED_FILES:
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+    case ADD_REFERENCED_FILES:
+    case REMOVE_REFERENCED_FILES:
+    case RECEIVE_CREATION_CONFIRM:
+    case RECEIVE_DELETION_CONFIRM:
+    case RECEIVE_ERROR:
+      const entity = action.entity
+      const endpointKey = entity
+        ? (action.relationName ? `${entity.type}/${entity.id}/${action.relationName}` : `${entity.type}`)
+        : action.doctype
+      return Object.assign({}, state, { [endpointKey]: endpoint(state[endpointKey] || {}, action) })
+    default:
+      return state
+  }
+}
+
+export default combineReducers({
+  schemas,
+  entities,
+  endpoints
+})
+
+// utils
+const removeObjectProperty = (obj, prop) => {
+  return Object.keys(obj).reduce((result, key) => {
+    if (key !== prop) {
+      result[key] = obj[key]
+    }
+    return result
+  }, {})
+}
+
+const objectifyEntitiesArray = (entities) => {
+  let obj = {}
+  entities.forEach(entity => { obj[entity.id] = entity })
+  return obj
+}
+
+const mapEntitiesToIds = (entities, doctype, ids) => {
+  if (!entities[doctype]) {
+    return ids.map(() => null)
```

Comment:
Is not that weird to receive an array of `length === ids.length` and content is `[null,null,null, ...]`?